cardano-api/src/Cardano/Api/IPC/Monad.hs

```diff
@@ -52,44 +59,56 @@ executeLocalStateQueryExpr
   -> Maybe ChainPoint
   -> LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO a
   -> IO (Either AcquiringFailure a)
-executeLocalStateQueryExpr connectInfo mpoint f = do
-  tmvResultLocalState <- newEmptyTMVarIO
+executeLocalStateQueryExpr connectInfo mpoint f =
+  runOopsInEither $ executeLocalStateQueryExpr_ connectInfo mpoint (lift f)
+
+-- | Execute a local state query expression.
+executeLocalStateQueryExpr_
+  :: forall e mode a . ()
+  => e `CouldBe` AcquiringFailure
+  => LocalNodeConnectInfo mode
+  -> Maybe ChainPoint
+  -> ExceptT (Variant e) (LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO) a
+  -> ExceptT (Variant e) IO a
```

Comment:
This function is slightly more interesting that the previous one.

We declare that `e ` could be `AcquiringFailure`.  There is nothing preventing `e` from also being some other error.  The function body doesn't throw anything other than `AcquiringFailure`, however, the function `f` can.  Any errors that `f` throws will propagate to the caller of `executeLocalStateQueryExpr_` for free.  Non-`oops` code will struggle to propagate it without introducing a convoluted type signature.

How about the non-`oops` wrapper?  That wrapper returns `IO (Either AcquiringFailure a)` and the function `f` it accepts is `LocalStateQueryExpr (BlockInMode mode) ChainPoint (QueryInMode mode) () IO a`.  It has no capacity to propagate errors.  It can only return values.